projects/packages/connection/src/class-authorize-json-api.php

```diff
@@ -41,7 +41,7 @@ class Authorize_Json_Api {
 	 */
 	public function verify_json_api_authorization_request( $environment = null ) {
 		$environment = $environment === null
-			? $_REQUEST // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- nonce verification handled later in function.
+			? $_REQUEST // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- nonce verification handled later in function. We need all of $_REQUEST in order to verify a cryptographic signature of the request data.
```

Comment:
Good catch, I missed that! Alternate approach in https://github.com/Automattic/jetpack/pull/37392/commits/b88ccaeae446a25efa040ce5dd992d6635064f30 so we use only the data we need. Feel free to suggest a better explanation if you think it's needed :)

Reply:
I, on the other side, like the idea of using exactly what we need from `$_REQUEST` just in case a monster is hidden there :)
For now I updated the code to do `array_intersect_key` as suggested. 
However, If you feel strongly about keeping it as is, I'm totally fine with that too :)